apps/framework-cli/src/cli/routines/ls.rs

```diff
@@ -348,3 +355,326 @@ fn format_topics(
+    fn to_json_string(&self) -> Result<String, Error> {
+        serde_json::to_string_pretty(&self)
+    }
+}
+
+impl From<FunctionProcess> for StreamTransformationInfo {
+    fn from(value: FunctionProcess) -> Self {
+        Self {
+            source: value.source_topic_id,
+            destinations: vec![value.target_topic_id],
+        }
+    }
+}
+
+#[derive(Debug, Serialize)]
+pub struct ResourceListing {
+    pub tables: Vec<TableInfo>,
+    pub streams: Vec<StreamInfo>,
+    pub ingestion_apis: Vec<IngestionApiInfo>,
+    pub sql_resources: Vec<SqlResourceInfo>,
+    pub consumption_apis: Vec<ConsumptionApiInfo>,
+    pub stream_transformations: Vec<StreamTransformationInfo>,
+}
+
+impl ResourceInfo for ResourceListing {
+    fn show(&self) {
+        self.tables.show();
+        self.streams.show();
+        self.ingestion_apis.show();
+        self.sql_resources.show();
+        self.consumption_apis.show();
+        self.stream_transformations.show();
+    }
+
+    fn to_json_string(&self) -> Result<String, Error> {
+        serde_json::to_string_pretty(&self)
+    }
+}
+
+pub async fn ls_dmv2(
+    project: &Project,
+    _type: Option<&str>,
+    name: Option<&str>,
+    json: bool,
+) -> Result<RoutineSuccess, RoutineFailure> {
+    let infra_map = InfrastructureMap::load_from_user_code(project)
+        .await
+        .map_err(|e| {
+            RoutineFailure::new(
+                Message {
+                    action: "Load".to_string(),
+                    details: "Infrastructure".to_string(),
+                },
+                e,
+            )
+        })?;
+
+    let (ingestion_apis, consumption_apis): (Vec<_>, Vec<_>) = infra_map
+        .api_endpoints
+        .values()
+        .filter(|api| name.is_none_or(|name| api.name.contains(name)))
+        .partition_map(to_info);
+    let resources = ResourceListing {
+        tables: infra_map
+            .tables
+            .into_values()
+            .filter(|api| name.is_none_or(|name| api.name.contains(name)))
+            .map(|t| t.into())
+            .collect(),
+        streams: infra_map
+            .topics
+            .into_values()
+            .filter(|api| name.is_none_or(|name| api.name.contains(name)))
+            .map(|t| StreamInfo::from_topic(t, &infra_map.topic_to_table_sync_processes))
+            .collect(),
+        ingestion_apis,
+        sql_resources: infra_map
+            .sql_resources
+            .into_values()
+            .filter(|api| name.is_none_or(|name| api.name.contains(name)))
+            .map(|resource| SqlResourceInfo {
+                name: resource.name,
+            })
+            .collect(),
+        consumption_apis,
+        stream_transformations: infra_map
+            .function_processes
+            .into_values()
+            .filter(|api| name.is_none_or(|name| api.name.contains(name)))
+            .map(|p| p.into())
+            .collect(),
+    };
+    let listing: &dyn ResourceInfo = match _type {
```

Comment:
Could they ls for multiple types?

Reply:
maybe, but probably worth merging this in now